src/main/kotlin/graphics/scenery/Spline.kt

```diff
@@ -0,0 +1,17 @@
+package graphics.scenery
+
+import org.joml.Vector3f
+
+/**
+ * This is the abstract class for all the spline classes currently existing. It takes a list
+ * of [controlPoints], which constrict spline. Moreover, one has to specify the number of points
+ * in one segment of the spline [n].
+ *
+ * @author Justin Bürger <burger@mpi-cbg.de>
+ */
+abstract class Spline(open val controlPoints: List<Vector3f>, open val n: Int) {
```

Comment:
Considering this doesn't really need to store state, maybe it would make more sense to make it an interface instead of an abstract class?